### PR TITLE
Support environment variables

### PR DESCRIPTION
**Pull request by punkupoz**
Support environment variables for configuration.

Env vars will overrides values in config file.

### CODE REVIEW

**Comment by andytruong** [comment]
> Support environment variables

We already support it (https://github.com/andytruong/redes-writer/blob/7.x/interfaces.go#L112-L116)

```go
func NewConfig(cnfPath string) (*Config, error) {
	yamlBytes, err := ioutil.ReadFile(cnfPath)
	expandedInput := os.ExpandEnv(string(yamlBytes))
```

**Comment by punkupoz** [comment]
> > Support environment variables
> 
> We already support it (https://github.com/andytruong/redes-writer/blob/7.x/interfaces.go#L112-L116)
> 
> ```go
> func NewConfig(cnfPath string) (*Config, error) {
> 	yamlBytes, err := ioutil.ReadFile(cnfPath)
> 	expandedInput := os.ExpandEnv(string(yamlBytes))
> ```

The ``os.ExpandEnv(string(yamlBytes))`` does so by replacing occurences of ${var} or $var in the configuration file.

The pull request support configurate with environment variable without having to write ${var} or $var into the configuration file.

File: config_test.go [hunk 1]
```diff
@@ -0,0 +1,123 @@
...
+	"github.com/stretchr/testify/assert"
+	"io/ioutil"
+	"os"
+	"testing"
+	"time"
+)
+
+func TestNewConfig(t *testing.T) {
+	assert := assert.New(t)
+	testConfName := "testConfig1.yaml"
```
Comment (andytruong) [inline config_test.go]: can we change this to `/tmp/redes-writer.qa.yaml`?

File: config.go [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+}
+
+// fileConfig receive a pointer to config and path to config file path
+// this function modify value in config pointer
+func fileConfig(cnf *Config, cnfPath string) error {
+	file, err := ioutil.ReadFile(cnfPath)
+	if err != nil {
+		return err
+	}
+	err = yaml.Unmarshal(file, cnf)
```
Comment (andytruong) [inline config.go]: We should keep os.EnvExpand() to not break old style config.

File: config_test.go [hunk 3]
```diff
@@ -0,0 +1,123 @@
...
+  bufferSize: 5000
+  flushInterval: 3s # for faster CI test running
+`)
+	envPrefix := "REDES_WRITER"
+
+	_ = os.Setenv(envPrefix+"_ADMIN_URL", "envenv:8484")
+	_ = os.Setenv(envPrefix+"_REDIS_URL", "redis://rere:6379?ssl=false")
+	_ = os.Setenv(envPrefix+"_ELASTICSEARCH_URL", "http://searchelast:9200/?sniff=true")
+
+	err := ioutil.WriteFile(testConfName, testFile, 0755)
```
Comment (andytruong) [inline config_test.go]: defer delete that temp file.

**Pull request merged**